app/src/main/java/app/gamenative/service/WishlistWebViewAdder.kt

```
@@ -0,0 +1,158 @@
+            fun finish(result: Boolean) {
+                if (done.isCompleted) return
+                done.complete(result)
+            }
+
+            webView.addJavascriptInterface(
+                object {
+                    @JavascriptInterface
+                    fun onResult(json: String) {
+                        val ok = try {
+                            JSONObject(json).optBoolean("success")
+                        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)